drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -309,8 +265,48 @@ impl ServerImpl {
         res
     }
 
-    // Fetch as many bytes as we can and parse the header.
+    fn do_send_request(&mut self, size: usize) -> Result<(), SprotError> {
+        ringbuf_entry!(Trace::SendRecv(size));
+        let buf = &self.tx_buf.as_slice()[..size];
+
+        // In order to improve reliability, start by sending only the
+        // first ROT_FIFO_SIZE bytes and then delaying a short time.
+        // If the RoT is ready, those first bytes will always fit
+        // in the RoT receive FIFO. Eventually, the RoT FW will respond
+        // to the interrupt and enter a tight loop to receive.
+        // The short delay should cover most of the lag in RoT interrupt
+        // handling.
+        let part1 = if let Some(part1) = buf.get(0..ROT_FIFO_SIZE.min(size)) {
```

Comment:
I know this is just moved code, but I believe you can get rid of the `if let` here and use indexing directly. You already know size fits in `buf` and you know `ROT_FIFO_SIZE` also fits in `buf`.

Reply:
done